src/component.jl

```diff
@@ -0,0 +1,185 @@
+        if it <= initial_time && (end_time <= (it + sys_resolution * sys_horizon)) &&
+                ((initial_time - it) % Dates.Millisecond(sys_resolution) == Dates.Second(0))
+            return get_forecast(T, _get_forecast_container(component), it, label)
+        end
+    end
+
+    throw(ArgumentError("did not find a forecast matching the requested parameters"))
+end
+
+function _get_forecast_container(component::InfrastructureSystemsType)
+    # TODO: The get__forecasts methods in PowerSystems need to be IS.get__forecasts.
+    #container = get__forecasts(component)
+    if :_forecasts in fieldnames(typeof(component))
+        container = component._forecasts
+    else
+        container = nothing
+    end
+    return container
+end
+
+function has_forecasts(component::InfrastructureSystemsType)
+    return !isnothing(_get_forecast_container(component))
+end
+
+function get_forecast_initial_times(
+                                    ::Type{T},
+                                    component::InfrastructureSystemsType,
+                                   ) where T <: Forecast
+    if !has_forecasts(component)
+        throw(ArgumentError("$(typeof(component)) does not have forecasts"))
+    end
+    return get_forecast_initial_times(forecast_external_to_internal(T),
+                                      _get_forecast_container(component))
+end
+
+function get_forecast_initial_times(
+                                    ::Type{T},
+                                    component::InfrastructureSystemsType,
+                                    label::AbstractString,
+                                   ) where T <: Forecast
+    if !has_forecasts(component)
+        throw(ArgumentError("$(typeof(component)) does not have forecasts"))
+    end
+    return get_forecast_initial_times(forecast_external_to_internal(T),
+                                      _get_forecast_container(component),
+                                      label)
+end
+
+function get_forecast_initial_times!(
+                                     initial_times::Set{Dates.DateTime},
+                                     component::InfrastructureSystemsType,
+                                    )
+    if !has_forecasts(component)
+        throw(ArgumentError("$(typeof(component)) does not have forecasts"))
+    end
+
+    get_forecast_initial_times!(initial_times, _get_forecast_container(component))
+end
+
+function get_forecast_keys(component::InfrastructureSystemsType)
+    return keys(_get_forecast_container(component).data)
+end
+
+function get_forecast_labels(
+                             ::Type{T},
+                             component::InfrastructureSystemsType,
+                             initial_time::Dates.DateTime,
+                            ) where T <: Forecast
+    return get_forecast_labels(forecast_external_to_internal(T), _get_forecast_container(component), initial_time)
+end
+
+function get_forecast_value(
```

Comment:
It will be better if we return 

`get_timeseries(forecast)[index]*get_field(component, label)`

Reply:
@claytonpbarrows @jd-lara 
The process that consumes the user descriptor file that maps user CSV column names to PowerSystem names is implemented in the PowerSystemTableData struct.  It is used during parsing but not normal operation.

Options:

1. Convert label during parsing only.  This means that any code that creates forecasts or calls add_forecast!(sys) would have to change the label being passed.
2. Restructure the code such that the System consumes the descriptors and performs the conversion.

I prefer #1 because it makes better sense to me and is much simpler.

Reply:
Also prefer 1.

Reply:
1 sounds good

Reply:
@jd-lara Would it be better to do this multiplication on the TimeArray as soon as we add the forecast to the component?  Doing what you suggested could mean that the user gets different values depending on what function they call.

Reply:
I don't think that doing the multiplication is the best option since the idea is that the user can manipulate the value that the scaling factor is being multiplied by. As for the consistency, we can change the naming to `get_scalingfactors` and return the time series and `get_forecast_value` that returns the value at the time for the label. This will avoid confusion.

Reply:
There is some nuance to this.  If the user calls get_forecast() then they will receive a Forecast object that has scaling factors (because we store scaling factors).  There is a get_forecast_value method on that object which just returns a value at an index - so that would return a scaling factor.  (But we could delete that method.)

If the user calls the get_forecast_value method that takes a Component then would could perform the multiplication.

Another option is to add an optional parameter to get_forecast to allow specification of scaling_factor or not.  Then the Forecast object they get back would have exactly what they want.

I think I like that best.  Here is what the functions would look like.  Let me know what you think.
```
get_forecast(T, sys, component, initial_time, label; scaling_factor=true)
get_forecast_value(T, sys, component, initial_time, label, index; scaling_factor=true)
get_forecast_value(forecast::Forecast, index)
get_data(forecast::Forecast) = forecast.data
get_time_series(forecast::Forecast) = get_data(forecast)
```